src/context/WishListContext.js

```diff
@@ -5,47 +5,89 @@ const WishListContext = createContext(null);
 const WishListProvider = ({children}) => {
     
     //-------------------for wishlist------------------------- 
-    const [ wishlist, setWishList ] = useState(JSON.parse(localStorage.getItem ("wishlist")));
+    const [ wishlist, setWishList ] = useState(localStorage.getItem ("wishlist")? JSON.parse(localStorage.getItem("wishlist")):[]);
 
-    //storing product in local storage 
+    //storing wishlist product in local storage 
     useEffect(()=> {
         localStorage.setItem("wishlist", JSON.stringify(wishlist))
     },[wishlist])
 
- 
     //using function adding product to wishlist
     const addToWishList = (product) => {
-        setWishList( prev => [...prev, product ] )
+
+        let isItemExist = wishlist.find(item => item._id === product._id)
+        
+        if(isItemExist){
+            setWishList(prev => prev.map(item => item._id === isItemExist._id ? {...item,quantity: item.quantity+1} : item) )
+        }
+        else{
+            setWishList(prev => [...prev, { ...product, quantity: 1 } ])
+        }
     }
     //using function removing product from wishlist
     const removeFromWishList = (id) => {
 
         setWishList(wishlistProducts => wishlistProducts.filter(item => item._id !== id)  )
     }
 
-
     //--------------------for AddToCart------------------------
-  
     const [cartItem , setcartItem] = useState(JSON.parse(localStorage.getItem ("cartItem")));
```

Comment:
It will be more scoring if you will use reducer instead state

Reply:
okay , I will try